### PR TITLE
ZCS-516 : Updated admin guide for Q1 2017 release

### PR DESCRIPTION
**Pull request by sneha-patil-synacor**
- Added S/MIME configuration changes
- Added SIEVE configuration changes

### CODE REVIEW

File: configuration.adoc [hunk 1]
```diff
@@ -1701,7 +1701,14 @@ zmcontrol restart
 S/MIME is a standard to send secure email messages. S/MIME messages use
 digital signature to authenticate and encrypt messages.
 
-=== Prerequisites
+Currently, there are two versions of S/MIME feature
```
Comment (fatkudu) [inline configuration.adoc]: Perhaps:
Currently, there are two different methods for providing the S/MIME feature

File: configuration.adoc [hunk 2]
```diff
@@ -1701,7 +1701,14 @@ zmcontrol restart
...
+
+. The old client based solution which requires Java 1.6 SE deployed on client machine
```
Comment (fatkudu) [inline configuration.adoc]: I think it read better with "the" in front of "client machine".
Same comment for the next line.

File: configuration.adoc [hunk 3]
```diff
@@ -1701,7 +1701,14 @@ zmcontrol restart
...
+. The new server based solution which does not require Java on client machine. The server performs all the cryptographic operations.
+
+=== Setting up client based S/MIME feature
```
Comment (fatkudu) [inline configuration.adoc]: Suggest:
Setting up for using the S/MIME feature using the client base solution.

File: configuration.adoc [hunk 4]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
 
 To set up another external LDAP server, click *Add Configuration*.
 
+=== Setting up server based S/MIME feature
```
Comment (fatkudu) [inline configuration.adoc]: Suggest:
Setting up for using the S/MIME feature using the server based solution.

Also recommend having this section first, as I suspect this is the preferred method?

File: configuration.adoc (hunk 2 above)
Comment (fatkudu) [inline configuration.adoc]: Is it worth guiding people towards the new method as the preferred method?

File: configuration.adoc [hunk 5]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+
+==== Prerequisites
+
+Same as the previous S/MIME version except that the Java is not required on client machine
```
Comment (fatkudu) [inline configuration.adoc]: Suggest:
Same as for the client based S/MIME solution except that Java is not required on the client machine

File: configuration.adoc [hunk 6]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+
+==== S/MIME License
+
+Same as the previous S/MIME version
```
Comment (fatkudu) [inline configuration.adoc]: Same as for the client based S/MIME solution?

File: configuration.adoc [hunk 7]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+
+==== Enable S/MIME Feature
+
+Same as the previous S/MIME version
+
+==== Importing S/MIME Certificates
+
+Same as the previous S/MIME version except that the recipients' public-key certificate no longer need to be stored into Local OS or browser keystore.
```
Comment (fatkudu) [inline configuration.adoc]: Suggest:
Same as for the client based S/MIME solution except that there is no need to store the recipients' public-key certificate in the Local OS or browser keystore.

File: configuration.adoc [hunk 8]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+The certificate can be published to all other places mentioned in previous S/MIME version.
+
+==== List of LDAP attributes introduced in new S\MIME feature
```
Comment (fatkudu) [inline configuration.adoc]: Suggest:
List of LDAP attributes introduced to support the server based S/MIME solution

File: configuration.adoc [hunk 9]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+
+. zimbraSmimeOCSPEnabled
+
+* Used by server at the time of validating the user as well as public certificates
+* If TRUE, the revocation check will be performed during certificate validation
+* If FALSE, the revocation check will not be performed during certificate validation
+
+. zimbraSmimePublicCertificateExtensions
+
+* The supported public certificate file extensions separated by comma
```
Comment (fatkudu) [inline configuration.adoc]: s/comma/commas/

File: configuration.adoc [hunk 10]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+* Contains the list of supported formats for userCertificate LDAP attribute
+* Default values: cer,crt,der,spc,p7b,p7r,sst,sto,pem
+* Zimbra web client retrieves from server the supported file formats/extensions for public certificate upload
```
Comment (fatkudu) [inline configuration.adoc]: Suggest:
Zimbra web client retrieves the supported file formats/extensions for public certificate upload from the server

File: configuration.adoc [hunk 11]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+
+. zimbraSmimeUserCertificateExtensions
+
+* The supported public certificate file extensions separated by comma.
```
Comment (fatkudu) [inline configuration.adoc]: s/comma/commas/

File: configuration.adoc [hunk 12]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+* Contains the list of supported formats for userSmimeCertificate LDAP attribute
```
Comment (fatkudu) [inline configuration.adoc]: s/for/for the/

File: configuration.adoc [hunk 13]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+
+The key names are:
+
+* mailboxd_truststore
+* mailboxd_truststore_password
+
+If the key is not defined in localconfig.xml, by default the values of mailboxd_truststore and mailboxd_truststore_password are:
+
+* <zimbra_java_home>/jre/lib/security/cacerts
+* changeit
```
Comment (fatkudu) [inline configuration.adoc]: I don't understand this.

File: configuration.adoc [hunk 14]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+. zimbraSmimeUserCertificateExtensions
+
+* The supported public certificate file extensions separated by comma.
+* Contains the list of supported formats for userSmimeCertificate LDAP attribute
+* Default values: p12,pfx
+* Zimbra web client retrieves from server the supported file formats/extensions for user certificate upload
+
+==== Process for Adding the CA certificate to the mailbox truststore for S/MIME
+
+S/MIME uses the mailbox trust store path and its password defined in localconfig.xml
```
Comment (fatkudu) [inline configuration.adoc]: Suggest "which are defined in localconfig.xml"

File: configuration.adoc [hunk 15]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+
+* mailboxd_truststore
+* mailboxd_truststore_password
+
+If the key is not defined in localconfig.xml, by default the values of mailboxd_truststore and mailboxd_truststore_password are:
+
+* <zimbra_java_home>/jre/lib/security/cacerts
+* changeit
+
+CA certificate can be imported to mailbox trust store by executing following command:
```
Comment (fatkudu) [inline configuration.adoc]: Suggest:
A CA certificate can be imported to the mailbox trust store by executing the following command:

File: configuration.adoc [hunk 16]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+
+keytool -import -alias <alias> -keystore <mailboxd_truststore path> -trustcacerts -file <CA_Cert>
```
Comment (fatkudu) [inline configuration.adoc]: Suggest using a source block for this:
[source,bash]
----
keytool -import -alias <alias> -keystore <mailboxd_truststore path> -trustcacerts -file <CA_Cert>
----

File: configuration.adoc [hunk 17]
```diff
@@ -1803,6 +1810,101 @@ Example, "userSMIMECertificate, UserCertificate"
...
+
+== Admin SIEVE Filter
+
+* The admin SIEVE feature enables the system administrator to define one SIEVE rule executed before and one SIEVE rule executed after the end user-defined filter rule.
+* The system-admin-defined SIEVE rule will be combined with the end-user-defined SEIVE rule, and then combined one big SIEVE rule will be applied to both incoming and outgoing messages. But, the scope of the variables is local within each script (zimbraMailAdminSieveScriptBefore, zimbraMailSieveScript, zimbraMailAdminSieveScriptAfter) OR (zimbraMailAdminOutgoingSieveScriptBefore, zimbraMailOutgoingSieveScript, zimbraMailAdminOutgoingSieveScriptAfter)
```
Comment (fatkudu) [inline configuration.adoc]: Suggest for the first sentence:
The system-admin-defined SIEVE rule will be combined with the end-user-defined SEIVE rule into a single rule. This rule will be applied to both incoming and outgoing messages

File: configuration.adoc (hunk 8 above)
Comment (plobbes) [inline configuration.adoc]: typo: S\MIME => S/MIME (backslash to forward slash)

File: configuration.adoc (hunk 17 above)
Comment (plobbes) [inline configuration.adoc]: typo: SEIVE => SIEVE

Comment (plobbes) [inline configuration.adoc]: Food for thought... should doc commits mix multiple/different new feature changes all in a single change set?  If we are disciplined with code changes to not mix things like this should we follow the same discipline for doc changes?

File: configuration.adoc (hunk 1 above)
Comment (sneha-patil-synacor) [inline configuration.adoc]: Fixed

File: configuration.adoc (hunk 13 above)
Comment (sneha-patil-synacor) [inline configuration.adoc]: Removed the pwd field value. Keeping only mailboxd_truststore value. Just to let the reader know the default truststore it is referring to.

File: configuration.adoc [hunk 18]
```diff
@@ -1803,6 +1810,103 @@ Example, "userSMIMECertificate, UserCertificate"
 
 To set up another external LDAP server, click *Add Configuration*.
 
+=== Setting up for using the S/MIME feature using the server base solution
```
Comment (fatkudu) [inline configuration.adoc]: "server based", not "server base"

**Comment by fatkudu** [comment]
Looks OK to me.  If possible, it would be nice to get reviews from other members of your team who understand the new code well?

File: configuration.adoc (hunk 18 above)
Comment (sneha-patil-synacor) [inline configuration.adoc]: Fixed

**Pull request merged**